client/state/selectors/test/has-cancelable-user-purchases.js

```diff
@@ -0,0 +1,109 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import hasCancelableUserPurchases from 'state/selectors/has-cancelable-user-purchases';
+
+describe( 'hasCancelableUserPurchases', () => {
+	const targetUserId = 123;
+	const examplePurchases = deepFreeze( [
+		{ ID: 1, product_name: 'domain registration', blog_id: 1337, user_id: targetUserId },
+		{ ID: 2, product_name: 'premium plan', blog_id: 1337, user_id: targetUserId },
+		{
+			ID: 3,
+			product_name: 'premium theme',
+			product_slug: 'premium_theme',
+			blog_id: 1337,
+			user_id: targetUserId,
+		},
+	] );
+
+	test( 'should return false because there are no purchases', () => {
+		const state = {
```

Comment:
For consistency and future proof, how about we `deepFreeze` all the `state` in all test cases like the first one?